Changelog.md

```diff
@@ -29,6 +29,7 @@ Bugfixes:
  * TypeChecker: Improved error message for constant variables with (nested) mapping types.
  * Yul Assembler: Fix internal error when function names are not unique.
  * Yul IR Generator: Do not output empty switches/if-bodies for empty contracts.
+ * ICE in StructType::isDynamicallyEncoded() when a function has a calldata struct argument with an internal type inside
```

Comment:
Does it happen for `calldata` structs or only for arrays of structs? Can you add a test case covering single structs (unless there's one already, please check)?

Reply:
@cameel  You've mentioned in #10516  that this works for a struct but not for an array of structs. 
This is because isDynamicallyEncoded does not get called in the first case case.

However, I've added a test case to check for a single struct. Will push in the next commit.